src/Collection.php

```diff
@@ -392,6 +396,55 @@ public function createIndexes(array $indexes, array $options = [])
         return $operation->execute($server);
     }
 
+    /**
+     * Create an Atlas Search index for the collection.
+     *
+     * @see https://www.mongodb.com/docs/manual/reference/command/createSearchIndexes/
+     * @see https://mongodb.com/docs/manual/reference/method/db.collection.createSearchIndex/
+     * @param string $name       List of search index specifications
+     * @param array  $definition Atlas Search index definition
+     * @param array  $options    Command options
+     * @return string The name of the created search index
+     * @throws UnsupportedException if options are not supported by the selected server
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function createSearchIndex(string $name, array $definition, array $options): string
+    {
+        $names = $this->createSearchIndexes([['name' => $name, 'definition' => $definition]], $options);
+
+        return current($names);
+    }
+
+    /**
+     * Create one or more Atlas Search indexes for the collection.
+     *
+     * Each element in the $indexes array must have a "name" and a "definition" document.
+     * For example:
+     *
+     *     $indexes = [
+     *         // Create a search index on all fields
+     *         [ 'name' => 'search_all', 'definition' => [ 'mappings' => [ 'dynamic' => true ] ] ],
+     *     ];
+     *
+     * @see https://www.mongodb.com/docs/manual/reference/command/createSearchIndexes/
+     * @see https://mongodb.com/docs/manual/reference/method/db.collection.createSearchIndex/
+     * @param array[] $indexes List of search index specifications
+     * @param array   $options Command options
+     * @return string[] The names of the created search indexes
+     * @throws UnsupportedException if options are not supported by the selected server
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function createSearchIndexes(array $indexes, array $options = []): array
```

Comment:
I tried to pass `timeoutMS: 1`, either the operation is super-fast, or the option is ignored.

Reply:
The PHP driver still doesn't support client-side operation timeouts (see: [PHPC-1760](https://jira.mongodb.org/browse/PHPC-1760)), so the option is ignored.

Anyway, you're adding `$options` here, so that will definitely cover us when support is added down the line. It does sound like the spec's omission of options was an oversight, though.

Reply:
Just had a separate thought. I wonder if `comment` is supported on these commands, since they are directed a mongod/mongos. It's not mentioned in the [syntax document](https://docs.google.com/document/d/1rCU7vLmMEn-87SuSZ3xnnSpnkwoWpekyOl15NPIqTJk/edit), but could be another case of the spec forgetting about a general option when defining the API.

Reply:
@GromNaN are we OK to resolve this thread?